### PR TITLE
PD-41 Data Structure Command History

### PR DESCRIPTION
**Pull request by crapper**

### CODE REVIEW

File: src/diagram/Field.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Field } from "./Field";
+
+test('Field constructor', () => {
+    const f = new Field("test", 1);
+    expect(f.name).toBe("test");
+    expect(f.length).toBe(1);
+});
+
+test('Field setName', () => {
```
Comment (Jerrylum) [inline src/diagram/Field.test.ts:9]: change the name to `Field name setter`

File: src/diagram/Field.test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const f = new Field("test", 1);
+    expect(f.name).toBe("test");
+    f.name = "test2";
+    expect(f.name).toBe("test2");
+});
+
+test('Field setLength', () => {
```
Comment (Jerrylum) [inline src/diagram/Field.test.ts:16]: change the name to `Field length setter`

File: src/config/Option.test.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
+import { CodePointBuffer, Parameter } from "../token/Tokens";
+import { BooleanOption, EnumOption, RangeOption } from "./Option";
+
+test("BooleanOption Get", () => {
```
Comment (Jerrylum) [inline src/config/Option.test.ts:4]: change to `BooleanOption get & getter function`

File: src/config/Option.test.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+  expect(option.defaultValue).toBe(true);
+  expect(option.getValue()).toBe(true);
+  expect(option.getUsageDescription()).toBe("TRUE | false");
+  const option2 = new BooleanOption("test", false);
+  expect(option2.defaultValue).toBe(false);
+  expect(option2.getValue()).toBe(false);
+  expect(option2.getUsageDescription()).toBe("true | FALSE");
+});
+
+test("BooleanOption Set Boolean", () => {
```
Comment (Jerrylum) [inline src/config/Option.test.ts:16]: change to `BooleanOption set & setter function`

**Review by Jerrylum — CHANGES_REQUESTED** [review]
Be careful when you name your test cases

File: src/diagram/Field.test.ts (hunk 1 above)
Comment (crapper) [reply src/diagram/Field.test.ts]: Updated, please refer to 
[c14362b](https://github.com/Jerrylum/protocol-diagram/pull/8/commits/c14362b46accd850aaa7de3765b4c67e42df8132)

File: src/diagram/Field.test.ts (hunk 2 above)
Comment (crapper) [reply src/diagram/Field.test.ts]: Updated, please refer to 
[c14362b](https://github.com/Jerrylum/protocol-diagram/pull/8/commits/c14362b46accd850aaa7de3765b4c67e42df8132)

File: src/config/Option.test.ts (hunk 3 above)
Comment (crapper) [reply src/config/Option.test.ts]: Updated, please refer to 
[c14362b](https://github.com/Jerrylum/protocol-diagram/pull/8/commits/c14362b46accd850aaa7de3765b4c67e42df8132)

File: src/config/Option.test.ts (hunk 4 above)
Comment (crapper) [reply src/config/Option.test.ts]: Updated, please refer to 
[c14362b](https://github.com/Jerrylum/protocol-diagram/pull/8/commits/c14362b46accd850aaa7de3765b4c67e42df8132)

**Review by Jerrylum — APPROVED** [review]
LGTM

**Pull request merged**